BUG: GMX tpr interm >= 103

tylerjereddy opened the pull request:
* Fixes gh-5145.

* Support for intermolecular bonded interactions (`interm` variable in our `.tpr` parsing code) was not added until GROMACS tpx 103 (release 5.1: https://manual.gromacs.org/5.1/ReleaseNotes/new-features.html#added-support-for-inter-molecular-bonded-interactions)

* So guard against reading that value in accordingly, and add a regression test for properly parsing one of the problematic `.tpr` files shared in the matching ticket.


## PR Checklist
<!-- Please use the following checklist to ensure the PR is ready to be reviewed/merged. -->
 - [x] Issue raised/referenced?
 - [x] Tests updated/added?
 - [ ] Documentation updated/added?
 - [x] `package/CHANGELOG` file updated?
 - [x] Is your name in `package/AUTHORS`? (If it is not, add it!)

## Developers Certificate of Origin
<!-- 
In order for this PR to be merged you **must certify that you are able to submit your code to be included in MDAnalysis**.
-->

I certify that I can submit this code contribution as described in the [**Developer Certificate of Origin**](https://developercertificate.org/), under the MDAnalysis [LICENSE](https://github.com/MDAnalysis/mdanalysis/blob/develop/LICENSE).


<!-- readthedocs-preview mdanalysis start -->
----
📚 Documentation preview 📚: https://mdanalysis--5146.org.readthedocs.build/en/5146/

<!-- readthedocs-preview mdanalysis end -->

tylerjereddy committed:
BUG: GMX tpr interm >= 103

* Fixes gh-5145.

* Support for intermolecular bonded interactions (`interm` variable in
our `.tpr` parsing code) was not added until GROMACS tpx 103 (release 5.1:
https://manual.gromacs.org/5.1/ReleaseNotes/new-features.html#added-support-for-inter-molecular-bonded-interactions)

* So guard against reading that value in accordingly, and add a
regression test for properly parsing one of the problematic `.tpr`
files shared in the matching ticket.

tylerjereddy commented on testsuite/MDAnalysisTests/data/281.tpr:
The user-provided `.tpr` is about 1 MB in size; we may want to consider using a smaller one with similar properties

orbeckst replied:
I agree – for this simple regression, tiny would be better as we're probably not re-using the file for anything else. Do you know how a smaller file could be generated?

orbeckst reviewed (changes requested):
Thanks @tylerjereddy , please add a CHANGELOG entry.

I agree that a smaller test file would be very good but at the end of the day I won't block over that.

tylerjereddy committed:
TST, DOC, MAINT: PR 5146 revisions

* Based on reviewer feedback, the `281.tpr` file used
for regression testing has been reduced in size--
it now only contains a single atom, and the filename was
changed to `281_small.tpr` to reflect that. Regression
testing was adjusted accordingly.

* The smaller TPR still causes the new regression test
case to fail if the source patch is reverted.

* `CHANGELOG` has been adjusted per reviewer request.

tylerjereddy commented:
@orbeckst ok, the most recent commit message summarizes the relevant changes made. I resolved the `CHANGELOG` merge conflicts as well (we should do something about that process at some point..).

Note that:
- the new TPR file, 1/3 the size of the original, had to be produced with a custom/ancient build of GROMACS `5.0.7`, using the procedure below the fold, in order to retain the actual reproduction of the original failure (mostly for my own reference, since I may need to do this again in the future)

<details>
<summary> Steps to reduce TPR file size while retaining appropriate properties to fail the new test </summary>

- using GROMACS `v5.0.7` (built from source on x86_64 Linux)
- `gmx make_ndx -f 281.tpr -o min.ndx`
- `a 1` (select a single atom)
- `q` (save it)
- `gmx convert-tpr -s 281.tpr -n min.ndx -o 281_small.tpr`
- select group `6`, with a single atom 

</details>

- since the new TPR only has a single atom, the regression test now checks for the same first/last position and velocity values; one could argue that retaining two atoms may be slightly preferable to ensure we get different first/last `x`/`v` values, but we have tons of test cases for that already and I decided to prioritize minimizing size to the extent possible
- of course, squash committing is important here, when ready, to avoid tracking *both* binary assets in the `git` history rather than just the small one

orbeckst reviewed (approved):
Thanks a lot for addressing all my comments and going to the length to produce a 5.0.7 minimal TPR file!
